leveldb/key.go

```diff
@@ -72,17 +72,17 @@ func init() {
 
 type internalKey []byte
 
-func makeInternalKey(dst, ukey []byte, seq uint64, kt keyType) internalKey {
+func makeInternalKey(dst, ukey []byte, seq uint64, kt keyType) (internalKey, error) {
 	if seq > keyMaxSeq {
-		panic("leveldb: invalid sequence number")
+		return nil, errors.New("leveldb: invalid sequence number")
```

Comment:
Could we include seq and keyMaxSeq in this error message too, in case we want it for troubleshooting later? `fmt.Errorf("leveldb: invalid sequence number (max %d but got %d)", keyMaxSeq, seq)` or similar